### PR TITLE
6th per claw mech

### PR DESCRIPTION
**Pull request by Ippyb**
6th period claw mech w/ servos :)

### CODE REVIEW

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -92,16 +96,31 @@ private void controllerBindings() {
...
+    }
+    if(controlXbox2.getBButtonReleased()){
+      if(clawSubsystem.clawIsLifted){
+        clawSubsystem.lowerClaw();
+      }
+      clawSubsystem.liftClaw();
+    }
+  };
+  tankSubsystem.setDefaultCommand(new RunCommand(tank, tankSubsystem));
+}
```
Comment (e3l) [inline src/main/java/frc/robot/RobotContainer.java]: idk what happened but the indenting here got real funky.

vscode has auto formatting too: https://code.visualstudio.com/docs/cpp/cpp-ide#_code-formatting

File: src/main/java/frc/robot/subsystems/claw/ClawSubsystem.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      public void closeClaw(){
+        clawIsOpen = false;  
+      }
+  
+      public void liftClaw(){
+       clawIsLifted = true;
+      }
+  
+      public void lowerClaw(){
+        clawIsLifted = true;
```
Comment (e3l) [inline src/main/java/frc/robot/subsystems/claw/ClawSubsystem.java]: should be false?

File: src/main/java/frc/robot/subsystems/claw/ClawSubsystem.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+      
+      @Override
+      public void periodic() {
+        if (clawIsOpen){
+          for (int i = 0; i < servos.length; i++){
+            servos[i].setAngle(openAngles[i]);
+          }
+        } 
+        else{
+          if (clawIsOpen){
```
Comment (e3l) [inline src/main/java/frc/robot/subsystems/claw/ClawSubsystem.java]: you can do two separate if loops so you don't have to duplicate logic, one for lift and one for claw :)

File: src/main/java/frc/robot/RobotContainer.java [hunk 4]
```diff
@@ -92,16 +96,31 @@ private void controllerBindings() {
...
         tankSubsystem.setCarDrivePowers(-controlXbox.getY(Hand.kLeft), controlXbox.getX(Hand.kRight));
-      } else {
-        tankSubsystem.setTankDrivePowers(-joystickLeft.getY(), -joystickRight.getY());
+    }
+    else {
+      tankSubsystem.setTankDrivePowers(-joystickLeft.getY(), -joystickRight.getY());
+    }
+
+   if(controlXbox2.getAButtonReleased()){
+      if(clawSubsystem.clawIsOpen){
```
Comment (e3l) [inline src/main/java/frc/robot/RobotContainer.java]: just being picky -- could do clawSubsystem.clawIsOpen = !clawSubsystem.clawIsOpen

**Comment by seasew** [comment]
Closed PR so we can delete this branch. Code has been moved to 2021ShopProj repo.

**Pull request closed without merging**